tests/test_app/test_composable.py

```diff
@@ -1118,7 +1119,7 @@ def test_apply_to_only_appends(half_dstore1, half_dstore2):
     process1 = reader1 + min_length1 + writer1
     # create paths as strings
     dstore1 = open_data_store(half_dstore1.source, suffix="fasta")
-    dstore1 = [str(m) for m in dstore1]
+    dstore1 = [str(Path(m.data_store.source) / m.unique_id) for m in dstore1]
```

Comment:
I have problem with this style, if users have to use this type of access to data_store members.

Reply:
They don't. This is just checking for the case where a string is provided in tests. A user should never need to do thjis since DataMember read will get them the data.